snap/container.go

```diff
@@ -33,7 +33,10 @@ import (
 // Container is the interface to interact with the low-level snap files
 type Container interface {
 	// ReadFile returns the content of a single file from the snap.
-	ReadFile(relative string) (content []byte, err error)
+	ReadFile(relative string) ([]byte, error)
+
+	// ReadDir returns the content of a single directory inside the snap.
+	ReadDir(path string) ([]os.FileInfo, error)
```

Comment:
I've been thinking some more about this. What if a snap author forgets to make the hook in question executable? Or has a directory in here instead? Wouldn't we want to know that when loading them info `snap.Info` and skip such hooks, or display a warning or error? Using `[]string` here makes that difficult, as I mentioned before. What do you think?